lib/config/specification.yml

```diff
@@ -1453,6 +1453,7 @@ project_accounting_records:
     - story_id
     - role_id
     - user_id
+    - external_reference
```

Comment:
Definitely not needed for this PR, but as a heads up, we will be adding more non-required fields to this table: https://kantata.atlassian.net/browse/OX-7402